crates/bindings/README.md

```diff
@@ -0,0 +1,517 @@
+
+The returned row has the `autoinc` column set to the value that was actually written into the database.
+
+```no_run
+#[table(name = example)]
+struct Example {
+    #[autoinc]
+    field: u32
+}
+
+#[reducer]
+fn insert_autoinc_example(ctx: &ReducerContext) {
+    for i in 1..=10 {
+        // These will have values of 1, 2, ..., 10
+        // at rest in the database, since they
+        // are inserted with the sentinel value 0.
+        let actual = ctx.db.example().insert(Example { field: 0 })
+        assert_eq!(actual.field, i);
+    }
+}
+```
+
+#### Indices
+
+SpacetimeDB supports both single- and multi-column indexes. Currently, only [B-Tree](https://en.wikipedia.org/wiki/B-tree) indexes are supported.
+
+Indexes are declared using the syntax:
+
+[`#[table(index(name = my_index, btree(columns = [a, b, c]))]`](macro@crate::table#index).
+
+Multiple indexes can be declared, separated by commas.
+
+Single-column indices can also be declared using the
+
+[`#[index(btree)]`](macro@crate::table#indexbtree)
+
+column attribute.
+
+Any index supports getting a [`BTreeIndex`] using [`ctx`](crate::ReducerContext)`.db.{table}().{index}()`. For example,
+    `ctx.db.people().name()`.
+    
+[`BTreeIndex`] provides:
+ - [`BTreeIndex::filter`]
+ - [`BTreeIndex::delete`]
+
+## Reducers
+
+Reducers are declared using the [`#[reducer]` macro](macro@crate::reducer).
+
+`#[reducer]` is always applied to top level Rust functions. Arguments of reducers must implement [`SpacetimeType`]. Reducers can either return nothing, or return a `Result<(), E>`, where `E` implements `Debug`.
+
+```rust
+use spacetimedb::{reducer, ReducerContext};
+
+#[derive(Debug)]
+enum GiveItemError {
+    NoSuchItem(u64),
+    NonTransferable(u64)
+}
+
+#[reducer]
+fn give_player_item(
+    ctx: &ReducerContext,
+    player_id: u64,
+    item_id: u64
+) -> Result<(), GiveItemError> {
+    /* ... */
+}
+```
+
+Every reducer runs inside a [database transaction](https://en.wikipedia.org/wiki/Database_transaction). <!-- TODO: specific transaction level guarantees. --> This means that reducers will not observe the effects of other reducers modifying the database while they run. Also, if a reducer fails, all of its changes to the database will automatically be rolled back. Reducers can fail by [panicking](::std::panic!) or by returning an `Err`.
+
+#### The `ReducerContext` Type
+
+Reducers have access to a special [`ReducerContext`] argument. This argument allows reading and writing the database attached to a module. It also provides some additional functionality, like generating random numbers and scheduling future operations.
+
+The most important field of [`ReducerContext`] is [`.db`](ReducerContext#structfield.db). This field provides a view of the module's database. The [`#[table]`](macro@crate::table) macro generates traits that add accessor methods to this field.
```

Comment:
"The most important field" seems like a silly thing to say, IMO. Does `ctx.sender` not matter? Should `ctx.identity()` just go screw itself?

```suggestion
[`ReducerContext`] provides access to the database tables via [the `.db` field](ReducerContext#structfield.db). The [`#[table]`](macro@crate::table) macro generates traits that add accessor methods to this field.
```

Reply:
ok but

```
james@DESKTOP-1O4DK4H MINGW64 /f/clockwork/SpacetimeDB/modules (jgilles/rust-docs-update)
$ rg 'ctx.db' --stats | grep matches
159 matches
8 files contained matches

james@DESKTOP-1O4DK4H MINGW64 /f/clockwork/SpacetimeDB/modules (jgilles/rust-docs-update)
$ rg 'ctx.sender' --stats | grep matches
14 matches
4 files contained matches

james@DESKTOP-1O4DK4H MINGW64 /f/clockwork/SpacetimeDB/modules (jgilles/rust-docs-update)
$ rg 'ctx.identity' --stats | grep matches
2 matches
1 files contained matches
```

:3